pallets/cash/src/internal/change_validators.rs

```diff
@@ -1,14 +1,43 @@
 use frame_support::storage::{IterableStorageMap, StorageMap};
 
-use crate::{reason::Reason, types::ValidatorKeys, Config, NextValidators};
+use crate::{
+    chains::{ChainHash, ChainId},
+    core::dispatch_notice_internal,
+    notices::{ChangeAuthorityNotice, Notice},
+    reason::Reason,
+    require,
+    types::ValidatorKeys,
+    Config, Event, Module, NextValidators, SessionInterface,
+};
 
 pub fn change_validators<T: Config>(validators: Vec<ValidatorKeys>) -> Result<(), Reason> {
```

Comment:
who calls change_validators?

Reply:
called here, must be root. so either sudo or by gov. https://github.com/compound-finance/compound-chain/blob/e17c2bad5ae7532ffe72f8ea9a1334b4e8fd3eac/pallets/cash/src/lib.rs#L447